sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -161,51 +162,45 @@ namespace Azure { namespace IO {
   };
 
   /**
-   * @brief #Azure::IO::BodyStream providing its data from a file.
+   * @brief #Azure::IO::BodyStream used for reading data from a file.
    */
   class FileBodyStream : public BodyStream {
   private:
     // immutable
-#if defined(AZ_PLATFORM_POSIX)
-    int m_fd;
-#elif defined(AZ_PLATFORM_WINDOWS)
-    HANDLE m_hFile;
+    Azure::IO::Internal::FileHandleHolder m_fileStreamHolder;
+#if defined(AZ_PLATFORM_WINDOWS)
+    HANDLE m_filehandle;
+#elif defined(AZ_PLATFORM_POSIX)
+    int m_fileDescriptor;
 #endif
-    int64_t m_baseOffset;
     int64_t m_length;
     // mutable
     int64_t m_offset;
+    BodyStream* m_parallelBodyStream;
```

Comment:
This is allowing any bodyStream to be here. If we are expecting `ParallelFileBodyStream` , then use that as the type for the pointer,

Reply:
We don't need to keep all bodyStreams in one hpp file,  We can have the base class here and each concrete class on its own file.
Then we can have a convenience header for end-user if they want to include all streams available from Core